includes/utilities.inc

```diff
@@ -388,10 +388,19 @@ function islandora_solr_prepare_solr_results($solr_results) {
         // Add link to search.
         if (in_array($field, $link_to_search)) {
           $map_to_link = function ($original_value, $formatted_value) use ($field) {
+            if (variable_get('islandora_solr_advanced_search_block_lucene_syntax_escape', FALSE)) {
+              $lucene_regex = variable_get('islandora_solr_advanced_search_block_lucene_regex_default', ISLANDORA_SOLR_QUERY_FACET_LUCENE_ESCAPE_REGEX_DEFAULT);
+              $linked_value = islandora_solr_facet_query_escape(stripcslashes($original_value), $lucene_regex);
```

Comment:
The `stripcslashes()` invocation makes a mess with the backslashes... though really, not sure we want to do the regex thing in this situation: The links we generate are for searches on the exact string value, right? The "regex doodad" is more to allow more permissive escaping in advanced search, intended to allow the use of more Lucene syntax, if desired.

Reply:
Similarly, the regex stuff is intended to pass the queries slightly different... not wrapping things in quotes: https://github.com/Islandora/islandora_solr_search/blob/53852cc5859edaba910229a3e31d6600f28b65a3/includes/blocks.inc#L761-L767